README.md

```diff
@@ -17,174 +21,66 @@ end
+}
 
-The `Number` module contains options that will validate Number data types. It converts "stringified" numerical values to numbers. It supports the following options:
-  * `:min` - Specifies the minimum value of the field.
-  * `:max` - Specifies the maximum value of the field.
-  * `:integer` - Specifies that the number must be an integer (no floating point). Allowed values are `true` and `false`. The default is `false`.
+Litmus.validate(params, schema)
+# => {:ok, %{"id" => 1, "new_user" => true, "pin" => 1234, "username" => "user@123", "account_ids" => [1, 3, 9]}}
 
-```
-iex> schema = %{
-...> "id" => %Litmus.Type.Number{
-...>   integer: true
-...> },
-...> "gpa" => %Litmus.Type.Number{
-...>   min: 0,
-...>   max: 4
-...>  }
-...> }
-iex> params = %{"id" => "123", "gpa" => 3.8}
-iex> Litmus.validate(params, schema)
-{:ok, %{"id" => 123, "gpa" => 3.8}}
-iex> params = %{"id" => "123.456", "gpa" => 3.8}
-iex> Litmus.validate(params, schema)
-{:error, "id must be an integer"}
+Litmus.validate(%{}, schema)
+# => {:error, "id is not allowed"}
 ```
 
-### Litmus.Type.String
+## Supported Types
 
-The `String` module contains options that will validate String data types. It converts boolean and number values to strings. It will also convert `nil` value to empty string. It supports the following options:
-  * `:min_length` - Specifies the minimum number of characters needed in the string. Allowed values are non-negative integers.
-  * `:max_length` - Specifies the maximum number of characters needed in the string. Allowed values are non-negative integers.
-  * `:length` - Specifies the exact number of characters needed in the string. Allowed values are non-negative integers.
-  * `:regex` - Specifies a Regular expression that a string must match. Allowed value is a struct consisting of `pattern` and `error_message`, where `pattern` is a `Regex` and `error_message` is a `String.t()` value. Default value for pattern is `nil`. If no error_message is given, the default message returned on error is `"#{field} must be in a valid format"`.
-  * `:trim` - Removes additional whitespaces in a string and returns the new value. Allowed values are `true` and `false`. The default is `false`.
+Litmus currently supports the following types.
 
-```
-iex> schema = %{
-...> "username" => %Litmus.Type.String{
-...>   min_length: 3,
-...>   max_length: 10,
-...>   trim: true
-...> },
-...> "password" => %Litmus.Type.String{
-...>   length: 6,
-...>   regex: %Litmus.Type.String.Regex{
-...>     pattern: ~r/^[a-zA-Z0-9_]*$/,
-...>     error_message: "password must be alphanumeric"
-...>   }
-...>  }
-...> }
-iex> params = %{"username" => " user123 ", "password" => "root01"}
-iex> Litmus.validate(params, schema)
-{:ok, %{"username" => "user123", "password" => "root01"}}
-iex> params = %{"username" => " user123 ", "password" => "ro!_@1"}
-iex> Litmus.validate(params, schema)
-{:error, "password must be alphanumeric"}
-```
+* `Litmus.Type.Any`
```

Comment:
Thoughts on linking the docs to each of these?

Reply:
So in hexdocs these are auto-linked:

![image](https://user-images.githubusercontent.com/1128750/44494888-81b4aa80-a622-11e8-88c9-3a75fd13303a.png)

I'm worried making them links in the raw readme will break those links. ANy clever ideas on what to do?